e2e/puppeteer-custom-environment.ts

```diff
@@ -21,24 +21,46 @@ class PuppeteerCustomEnvironment extends PuppeteerEnvironment {
   async handleTestEvent(event, state) {
     switch (event.name) {
     case 'test_fn_failure':
-      console.log('test failed: \t', event.test.name);
       if (state.currentlyRunningTest.invocations > retryTimes) {
-        const testName = state.currentlyRunningTest.name.replace(/\s/g, ''); // remove whitespaces
+        console.error(`Failed test:  "${event.test.name}"`);
+        const testName = fp.startCase(state.currentlyRunningTest.name).replace(/[^A-Z0-9]+/ig, '');
         const screenshotDir = 'logs/screenshot';
         await fs.ensureDir(screenshotDir);
           // move create-filename to helper.ts
         const timestamp = new Date().getTime();
-        const fileName = `${testName}_${timestamp}.png`
+        const fileName = `${testName}_${timestamp}.png`;
+
         const screenshotFile = `${screenshotDir}/${fileName}`;
         await this.global.page.screenshot({path: screenshotFile, fullPage: true});
-        console.error(`Test "${event.test.name}" failed. Saved screenshot ${screenshotFile}.`);
+        console.error(`Saved screenshot ${screenshotFile}`);
+
+        const htmlFileName = `${testName}_${timestamp}.html`;
+        await this.savePageToFile(htmlFileName);
```

Comment:
save both page html and take a screenshot when test fails.

Reply:
@als364 This change is it. Others are cleanup and improvement. 🙂Thanks for the quick turnaround.

Reply:
thank you!